### PR TITLE
fix autograd error and import error

### PR DESCRIPTION
**Pull request by al-jshen**
- fix autograd function error by replacing `Interp1d()(...)` with `interp1d(...)` (which calls `apply` instead of `forward`)
Screen Shot 2022-12-14 at 11 20 22 PM

- fix import error in `README.md` by exporting `SDSS` and `BOSS` in `data/__init__.py`
- add `.gitignore`, run `black` and `isort`

### CODE REVIEW

**Comment by pmelchior** [comment]
starting with a PR, excellent!

What version of pytorch are you on? I haven't seen this error so far.

File: spender/data/__init__.py [hunk 1]
```diff
@@ -0,0 +1 @@
+from .sdss import BOSS, SDSS
```
Comment (pmelchior) [inline spender/data/__init__.py:1]: There's a reason I left this one out. The data handling of different survey will/might require special packages, so I don't want to force everyone to install those just to get one survey to load. Please remove this one.

**Comment by al-jshen** [comment]
> What version of pytorch are you on? I haven't seen this error so far.

I'm on Pytorch 1.13.1 (the latest stable version), but based on [this](https://discuss.pytorch.org/t/legacy-autograd-function-with-non-static-forward-method-is-deprecated-and-will-be-removed-in-1-3/69869) it seems like the error has been there for at least 2.5 years now (the fix is also in that thread).

File: spender/data/__init__.py (hunk 1 above)
Comment (al-jshen) [reply spender/data/__init__.py]: removed, and updated README.md example

**Review by pmelchior — APPROVED** [review]
this solve the RunTimeError; good catch in the README

**Pull request merged**